components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+        hintText = hintText,
+        hintContentDescription = hintContentDescription,
+        placeholderText = placeholderText,
+        errorText = errorText,
+        errorContentDescription = errorContentDescription,
+        showButtonContentDescription = showButtonContentDescription,
+        hideButtonContentDescription = hideButtonContentDescription,
+        showButtonStateDescription = showButtonStateDescription,
+        hideButtonStateDescription = hideButtonStateDescription,
+        maxChars = maxChars,
+        singleLine = singleLine,
+        keyboardOptions = KeyboardOptions(
+            keyboardType = if (numericOnly) KeyboardType.NumberPassword else KeyboardType.Password
+        ),
+        interactionSource = interactionSource,
+        requiredSequencesSpacing = requiredSequencesSpacing,
+        textStyle = textStyle
+    )
+}
+
+@Composable
+fun animateBorderStrokeAsState(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp,
+    unfocusedBorderThickness: Dp
+): State<BorderStroke> {
+    val focused by interactionSource.collectIsFocusedAsState()
+    val indicatorColor = indicatorColor(enabled, isError, interactionSource)
+    val targetThickness = if (focused) focusedBorderThickness else unfocusedBorderThickness
+    val animatedThickness = if (enabled) {
+        animateDpAsState(targetThickness, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(unfocusedBorderThickness)
+    }
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value))
+    )
+}
+
+var borderStroke: State<BorderStroke>? = null
+
+@Composable
+fun CalculateBorder(
+    enabled: Boolean = true,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp = FocusedBorderThickness,
+    unfocusedBorderThickness: Dp = UnfocusedBorderThickness
+) {
+    borderStroke = animateBorderStrokeAsState(
+        enabled,
+        isError,
+        interactionSource,
+        focusedBorderThickness,
+        unfocusedBorderThickness
+    )
+}
+
+@Composable
+fun indicatorColor(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource
+): State<Color> {
+    val focused by interactionSource.collectIsFocusedAsState()
+
+    val colors = HmrcTheme.colors
+
+    val focusedBorderColor = colors.hmrcBlue
+    val errorBorderColor = colors.hmrcBlack
+    val unfocusedBorderColor = colors.hmrcBlack
+
+    val targetValue = when {
+        isError -> errorBorderColor
+        focused -> focusedBorderColor
+        else -> unfocusedBorderColor
+    }
+    return if (enabled) {
+        animateColorAsState(targetValue, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(targetValue)
+    }
+}
+
+@SuppressWarnings("LongMethod", "ComplexMethod")
+@Composable
+fun PasswordTextInputView(
```

Comment:
```suggestion
private fun PasswordTextInputView(
```

Reply:
I don't think we want to expose this part to the consumer so should be private :)